src/brasil/gov/portlets/tests/test_setup.py

```diff
@@ -42,23 +45,88 @@ def test_browser_layer_installed(self):
     def test_version(self):
         self.assertEqual(
             self.st.getLastVersionForProfile(self.profile),
-            (u'1000',)
+            (u'1001',)
         )
 
 
 class TestUpgrade(BaseTestCase):
     """Ensure product upgrades work."""
 
-    def test_to1000_available(self):
-
+    def list_upgrades(self, source, destination):
         upgradeSteps = listUpgradeSteps(self.st,
                                         self.profile,
-                                        'unknown')
+                                        source)
+        if source == '0':
+            source = (source, '0')
+        else:
+            source = (source, )
+
         step = [step for step in upgradeSteps
-                if (step[0]['dest'] == ('1000',))
-                and (step[0]['source'] == ('unknown',))]
+                if (step[0]['dest'] == (destination,))
+                and (step[0]['source'] == source)]
+        return step
+
+    def execute_upgrade(self, source, destination):
+        # Setamos o profile para versao source
+        self.st.setLastVersionForProfile(self.profile, source)
+
+        # Pegamos os upgrade steps
+        upgradeSteps = listUpgradeSteps(self.st,
```

Comment:
follow conventions for [breaking lines](http://docs.plone.org/external/plone.api/docs/contribute/conventions.html#breaking-lines) here and all over the code